[css-ui] Fix box-sizing web platform test failures

karlcow opened the pull request:
#### 21c66fcd81750f21944d2057c20c00d2633011b6
<pre>
[css-ui] Fix box-sizing web platform test failures
<a href="https://bugs.webkit.org/show_bug.cgi?id=175287">https://bugs.webkit.org/show_bug.cgi?id=175287</a>
rdar://103700285

Reviewed by Tim Nguyen.

This fixes the logicalWidth computation in RenderReplaced according
to the equation described in the CSS 2.2 specification
<a href="https://www.w3.org/TR/CSS22/visudet.html#blockwidth">https://www.w3.org/TR/CSS22/visudet.html#blockwidth</a>

This also removes two tests and their expectations from the tree
which are covered by WPT tests and have also been removed
from the blink codebase. That would fix
<a href="https://bugs.webkit.org/show_bug.cgi?id=228211">https://bugs.webkit.org/show_bug.cgi?id=228211</a>

* LayoutTests/css2.1/20110323/replaced-intrinsic-003.htm: Removed.
* LayoutTests/css2.1/20110323/resources/replaced-intrinsic-003.svg: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/position-absolute-replaced-minmax-expected.txt:
* LayoutTests/platform/glib/css2.1/20110323/replaced-intrinsic-003-expected.txt: Removed.
* LayoutTests/platform/gtk/css2.1/20110323/replaced-intrinsic-003-expected.png: Removed.
* LayoutTests/platform/gtk/svg/hixie/intrinsic/003-expected.png: Removed.
* LayoutTests/platform/gtk/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/svg/hixie/intrinsic/003-expected.png: Removed.
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/ios/css2.1/20110323/replaced-intrinsic-003-expected.txt: Removed.
* LayoutTests/platform/ios/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-ventura-wk2-lbse-text/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/mac-ventura-wk2-pixel/svg/hixie/intrinsic/003-expected.png: Removed.
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac/css2.1/20110323/replaced-intrinsic-003-expected.png: Removed.
* LayoutTests/platform/mac/css2.1/20110323/replaced-intrinsic-003-expected.txt: Removed.
* LayoutTests/platform/mac/svg/hixie/intrinsic/003-expected.png: Removed.
* LayoutTests/platform/mac/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/win/css2.1/20110323/replaced-intrinsic-003-expected.txt: Removed.
* LayoutTests/platform/win/svg/hixie/intrinsic/003-expected.png: Removed.
* LayoutTests/platform/win/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/win/TestExpectations:
* LayoutTests/platform/wincairo/css2.1/20110323/replaced-intrinsic-003-expected.txt: Removed.
* LayoutTests/platform/wincairo/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/svg/hixie/intrinsic/003.html: Removed.
* LayoutTests/svg/hixie/intrinsic/resources/003.svg: Removed.
* LayoutTests/TestExpectations:
* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::RenderReplaced::computeConstrainedLogicalWidth const):

Canonical link: <a href="https://commits.webkit.org/258342@main">https://commits.webkit.org/258342@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dac158050e19c0108801855df1fc3a83ceb964d9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101561 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10716 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34620 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110864 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/171093 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/105541 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11671 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1618 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93954 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108652 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107343 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8892 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92138 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/35417 "Passed tests") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/90786 "Found 1 new API test failure: TestWebKitAPI.WebKit.QuotaDelegateReload (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23554 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78415 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4314 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25056 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4383 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1512 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10469 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44544 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5719 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6143 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/TestExpectations:
```diff
@@ -3128,13 +3128,13 @@ webkit.org/b/172609 fetch/closing-while-fetching-blob.html [ Skip ]
+webkit.org/b/175287 imported/w3c/web-platform-tests/css/css-ui/box-sizing-018.html [ Pass ]
+webkit.org/b/175287 imported/w3c/web-platform-tests/css/css-ui/box-sizing-019.html [ Pass ]
+webkit.org/b/175287 imported/w3c/web-platform-tests/css/css-ui/box-sizing-024.html [ Pass ]
+webkit.org/b/175287 imported/w3c/web-platform-tests/css/css-ui/box-sizing-025.html [ Pass ]
```
Removing these lines would be better than marking them as `Pass`

karlcow replied:
done

nt1m commented on LayoutTests/TestExpectations:
```diff
@@ -5938,10 +5938,10 @@ imported/w3c/web-platform-tests/url/idlharness-shadowrealm.window.html [ Pass Fa
+imported/w3c/web-platform-tests/css/css-typed-om/rotate-by-added-angle.html [ Pass ]
+imported/w3c/web-platform-tests/css/css-typed-om/width-by-clamp-px-em.html [ Pass ]
+imported/w3c/web-platform-tests/css/css-typed-om/width-by-max-px-em.html [ Pass ]
+imported/w3c/web-platform-tests/css/css-typed-om/width-by-min-px-em.html [ Pass ]
```
ditto

karlcow replied:
done

nt1m commented on LayoutTests/TestExpectations:
```diff
@@ -6200,3 +6200,7 @@ imported/w3c/web-platform-tests/css/css-nesting/conditional-properties.html [ Im
+
+# To be removed?
+webkit.org/b/228211 css2.1/20110323/replaced-intrinsic-003.htm [ Failure ]
+webkit.org/b/228211 svg/hixie/intrinsic/003.html [ Failure ]
```
Probably makes sense to remove these tests from the tree. As your passes show, there are other tests covering this. They have been removed from the blink codebase as well (cs.chromium.org)

karlcow replied:
done

karlcow committed:
[css-ui] Fix box-sizing web platform test failures
https://bugs.webkit.org/show_bug.cgi?id=175287
rdar://103700285

Reviewed by Tim Nguyen.

This fixes the logicalWidth computation in RenderReplaced according
to the equation described in the CSS 2.2 specification
https://www.w3.org/TR/CSS22/visudet.html#blockwidth

This also removes two tests and their expectations from the tree
which are covered by WPT tests and have also been removed
from the blink codebase. That would fix
https://bugs.webkit.org/show_bug.cgi?id=228211

* LayoutTests/css2.1/20110323/replaced-intrinsic-003.htm: Removed.
* LayoutTests/css2.1/20110323/resources/replaced-intrinsic-003.svg: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/position-absolute-replaced-minmax-expected.txt:
* LayoutTests/platform/glib/css2.1/20110323/replaced-intrinsic-003-expected.txt: Removed.
* LayoutTests/platform/gtk/css2.1/20110323/replaced-intrinsic-003-expected.png: Removed.
* LayoutTests/platform/gtk/svg/hixie/intrinsic/003-expected.png: Removed.
* LayoutTests/platform/gtk/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/svg/hixie/intrinsic/003-expected.png: Removed.
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/ios/css2.1/20110323/replaced-intrinsic-003-expected.txt: Removed.
* LayoutTests/platform/ios/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-ventura-wk2-lbse-text/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/mac-ventura-wk2-pixel/svg/hixie/intrinsic/003-expected.png: Removed.
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac/css2.1/20110323/replaced-intrinsic-003-expected.png: Removed.
* LayoutTests/platform/mac/css2.1/20110323/replaced-intrinsic-003-expected.txt: Removed.
* LayoutTests/platform/mac/svg/hixie/intrinsic/003-expected.png: Removed.
* LayoutTests/platform/mac/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/win/css2.1/20110323/replaced-intrinsic-003-expected.txt: Removed.
* LayoutTests/platform/win/svg/hixie/intrinsic/003-expected.png: Removed.
* LayoutTests/platform/win/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/win/TestExpectations:
* LayoutTests/platform/wincairo/css2.1/20110323/replaced-intrinsic-003-expected.txt: Removed.
* LayoutTests/platform/wincairo/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/hixie/intrinsic/003-expected.txt: Removed.
* LayoutTests/svg/hixie/intrinsic/003.html: Removed.
* LayoutTests/svg/hixie/intrinsic/resources/003.svg: Removed.
* LayoutTests/TestExpectations:
* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::RenderReplaced::computeConstrainedLogicalWidth const):

Canonical link: https://commits.webkit.org/258342@main
